### ISSUE TITLE
[P2][M6] MCP server: integration tests with end-to-end protocol verification

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#100]
## Problem

After #98 and #99, the MCP code generator produces C source and compiles it, but there are no automated tests that verify the generated server actually works end-to-end — correct MCP protocol responses, correct function dispatch, correct argument parsing, correct result formatting.

Unit tests in `mcp_gen.rs` verify the generated C source contains expected strings, but don't verify the compiled server behaves correctly over stdio. A single typo in a format string, a missing null terminator, or an off-by-one in JSON parsing could produce a server that compiles but returns garbage.

## Expected behavior

Tests should:
1. Generate an MCP wrapper for known C functions
2. Compile the wrapper to a binary
3. Send MCP protocol messages to the binary's stdin
4. Verify the JSON-RPC responses match expected output
5. Cover: initialize, tools/list, tools/call with various types, error cases

## Reproduction

Currently there is no automated way to verify the generated MCP server produces correct output. Manual testing is required: build a server, pipe JSON to it, eyeball the results. This is fragile and doesn't catch regressions.

## Suggested approach

Create an integration test module that:

1. **Uses `generate_mcp_wrapper()` directly** to produce the C source (doesn't need `ccc --mcp`)
2. **Writes the C source to a temp file** and compiles with `cc`
3. **Spawns the compiled binary** as a child process
4. **Sends JSON-RPC messages** to stdin and reads responses from stdout
5. **Parses and verifies** the JSON responses

```rust
fn build_and_query(c_code: &str, messages: &[&str]) -> Vec<String> {
    let sema = make_sema(c_code);
    let wrapper = generate_mcp_wrapper(&sema, &write_temp_c(c_code));
    let binary = compile_wrapper(&wrapper);
    let mut child = Command::new(&binary).stdin(Pipe).stdout(Pipe).spawn().unwrap();
    // Send messages, collect responses
}
```

**Test cases:**

- `initialize` → verify protocolVersion, capabilities, serverInfo
- `tools/list` → verify function names, param schemas, required fields
- `tools/call` with int params → verify correct integer result
- `tools/call` with double params → verify correct float result
- `tools/call` with string params → verify correct string handling
- `tools/call` with unknown tool → verify error response
- Invalid method → verify "Method not found" error
- Empty input line → verify no crash (skip gracefully)

## Files to modify

- `src/mcp_gen.rs` — Add `#[cfg(test)] mod integration_tests` section with the test infrastructure. These tests are heavier than unit tests (they compile and spawn binaries) so they should be clearly separated.

## Tests

```rust
#[cfg(test)]
mod integration_tests {
    use std::process::{Command, Stdio};
    use std::io::{Write, BufRead, BufReader};

    #[test]
    fn initialize_response() {
        let responses = build_and_query(
            "int add(int a, int b) { return a + b; }",
            &[r#"{"jsonrpc":"2.0","id":1,"method":"initialize","params":{}}"#],
        );
        assert!(responses[0].contains("protocolVersion"));
        assert!(responses[0].contains("ccc-mcp"));
    }

    #[test]
    fn tools_list_shows_functions() {
        let responses = build_and_query(
            "int add(int a, int b) { return a + b; } double mul(double x, double y) { return x*y; }",
            &[
                r#"{"jsonrpc":"2.0","id":1,"method":"initialize","params":{}}"#,
                r#"{"jsonrpc":"2.0","id":2,"method":"tools/list","params":{}}"#,
            ],
        );
        assert!(responses[1].contains("\"add\""));
        assert!(responses[1].contains("\"mul\""));
    }

    #[test]
    fn tool_call_returns_correct_result() {
        let responses = build_and_query(
            "int add(int a, int b) { return a + b; }",
            &[
                r#"{"jsonrpc":"2.0","id":1,"method":"initialize","params":{}}"#,
                r#"{"jsonrpc":"2.0","id":2,"method":"tools/call","params":{"name":"add","arguments":{"a":3,"b":4}}}"#,
            ],
        );
        assert!(responses[1].contains("\"7\""));
    }
}
```

## Acceptance criteria

- [ ] Integration test infrastructure compiles C and spawns the binary
- [ ] `initialize` response verified (protocol version, server info)
- [ ] `tools/list` response verified (function names and schemas present)
- [ ] `tools/call` with integer params returns correct result
- [ ] `tools/call` with double params returns correct result (within precision)
- [ ] Error cases tested (unknown tool, unknown method)
- [ ] Tests pass on macOS (using host `cc` for compilation)
- [ ] Clean build: `cargo build --release`
- [ ] All existing tests pass: `cargo test --lib`

Part of [MILESTONE] M6: AI-Native Tooling — MCP Server Generation (#97)
